spec/datadog/profiling/collectors/cpu_and_wall_time_spec.rb

```diff
@@ -43,22 +43,31 @@
   end
 
   describe '#sample' do
+    let(:pprof_result) do
+      serialization_result = recorder.serialize
+      raise 'Unexpected: Serialization failed' unless serialization_result
+
+      serialization_result.last
+    end
+    let(:samples) { samples_from_pprof(pprof_result) }
+
     it 'samples all threads' do
       all_threads = Thread.list
 
-      decoded_profile = sample_and_decode
+      cpu_and_wall_time_collector.sample
+      samples = samples_from_pprof(pprof_result)
 
-      expect(decoded_profile.sample.size).to be all_threads.size
+      expect(Thread.list).to eq(all_threads), 'Threads finished during this spec, causing flakiness!'
+      expect(samples.size).to be all_threads.size
     end
 
-    def sample_and_decode
+    it 'tags the samples with the object ids of the Threads they belong to' do
       cpu_and_wall_time_collector.sample
 
-      serialization_result = recorder.serialize
-      raise 'Unexpected: Serialization failed' unless serialization_result
+      samples = samples_from_pprof(pprof_result)
```

Comment:
Just noticed I left this `samples = ...` that is redundant with the `let` above. I'm already at work preparing another PR that also changes this file, so I'll include that cleanup in the next PR.